lib/barlix/gs1.ex

```diff
@@ -0,0 +1,190 @@
+  defp encoding(57), do: [1, 1, 1, 0, 1, 1, 0, 1, 0, 0, 0]
+  defp encoding(58), do: [1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 0]
+  defp encoding(59), do: [1, 1, 1, 0, 0, 0, 1, 1, 0, 1, 0]
+  defp encoding(60), do: [1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 0]
+  defp encoding(61), do: [1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0]
+  defp encoding(62), do: [1, 1, 1, 1, 0, 0, 0, 1, 0, 1, 0]
+  defp encoding(63), do: [1, 0, 1, 0, 0, 1, 1, 0, 0, 0, 0]
+  defp encoding(64), do: [1, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0]
+  defp encoding(65), do: [1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0]
+  defp encoding(66), do: [1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0]
+  defp encoding(67), do: [1, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0]
+  defp encoding(68), do: [1, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0]
+  defp encoding(69), do: [1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0]
+  defp encoding(70), do: [1, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0]
+  defp encoding(71), do: [1, 0, 0, 1, 1, 0, 1, 0, 0, 0, 0]
+  defp encoding(72), do: [1, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0]
+  defp encoding(73), do: [1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0]
+  defp encoding(74), do: [1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0]
+  defp encoding(75), do: [1, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0]
+  defp encoding(76), do: [1, 1, 0, 0, 1, 0, 1, 0, 0, 0, 0]
+  defp encoding(77), do: [1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 0]
+  defp encoding(78), do: [1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0]
+  defp encoding(79), do: [1, 0, 0, 0, 1, 1, 1, 1, 0, 1, 0]
+  defp encoding(80), do: [1, 0, 1, 0, 0, 1, 1, 1, 1, 0, 0]
+  defp encoding(81), do: [1, 0, 0, 1, 0, 1, 1, 1, 1, 0, 0]
+  defp encoding(82), do: [1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0]
+  defp encoding(83), do: [1, 0, 1, 1, 1, 1, 0, 0, 1, 0, 0]
+  defp encoding(84), do: [1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0]
+  defp encoding(85), do: [1, 0, 0, 1, 1, 1, 1, 0, 0, 1, 0]
+  defp encoding(86), do: [1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 0]
+  defp encoding(87), do: [1, 1, 1, 1, 0, 0, 1, 0, 1, 0, 0]
+  defp encoding(88), do: [1, 1, 1, 1, 0, 0, 1, 0, 0, 1, 0]
+  defp encoding(89), do: [1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0]
+  defp encoding(90), do: [1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 0]
+  defp encoding(91), do: [1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 0]
+  defp encoding(92), do: [1, 0, 1, 0, 1, 1, 1, 1, 0, 0, 0]
+  defp encoding(93), do: [1, 0, 1, 0, 0, 0, 1, 1, 1, 1, 0]
+  defp encoding(94), do: [1, 0, 0, 0, 1, 0, 1, 1, 1, 1, 0]
+  defp encoding(95), do: [1, 0, 1, 1, 1, 1, 0, 1, 0, 0, 0]
+  defp encoding(96), do: [1, 0, 1, 1, 1, 1, 0, 0, 0, 1, 0]
+  defp encoding(97), do: [1, 1, 1, 1, 0, 1, 0, 1, 0, 0, 0]
+  defp encoding(98), do: [1, 1, 1, 1, 0, 1, 0, 0, 0, 1, 0]
+  defp encoding(99), do: [1, 0, 1, 1, 1, 0, 1, 1, 1, 1, 0]
+  defp encoding(100), do: [1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 0]
+  defp encoding(101), do: [1, 1, 1, 0, 1, 0, 1, 1, 1, 1, 0]
+  defp encoding(102), do: [1, 1, 1, 1, 0, 1, 0, 1, 1, 1, 0]
+  defp encoding(103), do: [1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0]
+  defp encoding(104), do: [1, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0]
+  defp encoding(105), do: [1, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0]
```

Comment:
Perhaps worth the addition of a comment (or even just a mention in the commit) referencing the source for the encoding patterns for the convenience of future spelunkers? I realise it's somewhat implied by the module doc, but it can't hurt to add some additional guidance.
i.e. https://en.wikipedia.org/wiki/Code_128#Bar_code_widths or similar..

Reply:
Yeah good point